java/arcs/core/data/testdata/Manifest2ProtoTest.textproto

```diff
@@ -18,71 +18,79 @@ recipes {
     type {
       collection {
         collection_type {
-          entity {
-            schema {
-              names: "Product"
-              names: "Thing"
-              fields {
-                key: "name"
-                value {
-                  primitive: TEXT
-                }
-              }
-              fields {
-                key: "dimensions"
-                value {
-                  collection {
-                    collection_type {
-                      primitive: NUMBER
+          variable {
```

Comment:
I don't think we want that to happen here.

While all handle types are type variables in a recipe, as they are the result of type inference, we want the proto representation of the recipe handle to be a concrete type, which is a resolution of the typevar.

How do we do that?

Strawperson: In the manifest2proto we check if the type variable is resolved, and if it is resolved we encode its resolution, otherwise we encode the type variable. We should also do maybeEnsureResolved() on the handle type in the AllocatorRecipeResolver to force the type ranges to become concrete types.

Reply:
Is it ever the case that the type of handle can be typevar in a resolved recipe?

Reply:
Well... it can be range, but in such case our customer wants the minimum.

I think for foreseeable future we should write the resolved type and not worry too much about our ambitions that all handles are type ranges, or even more ambitiously type families (via type adapters).